beacon-chain/operations/service.go

```diff
@@ -235,10 +238,16 @@ func (s *Service) handleProcessedBlock(_ context.Context, message proto.Message)
 // removePendingAttestations removes a list of attestations from DB.
 func (s *Service) removePendingAttestations(attestations []*pb.Attestation) error {
 	for _, attestation := range attestations {
-		if err := s.beaconDB.DeleteAttestation(attestation); err != nil {
+		hash, err := hashutil.HashProto(attestation)
+		if err != nil {
 			return err
 		}
-		log.WithField("blockRoot", fmt.Sprintf("0x%x", attestation.Data.BeaconBlockRootHash32)).Info("Attestation removed")
+		if s.beaconDB.HasAttestation(hash) {
+			if err := s.beaconDB.DeleteAttestation(attestation); err != nil {
+				return err
+			}
+			log.WithField("blockRoot", fmt.Sprintf("0x%x", attestation.Data.BeaconBlockRootHash32)).Info("Attestation removed")
```

Comment:
I still think these logs with hashes suck - they don't tell us much at all and are not even useful for debugging, I vote to eliminate this log completely

Reply:
Also, you dont need to prefix with `0x` if you use `%#x`

Reply:
fixed